homeassistant/components/music_assistant/media_player.py

```diff
@@ -0,0 +1,729 @@
+            self._attr_media_title = player.current_media.title
+            self._attr_media_artist = player.current_media.artist
+            self._attr_media_album_name = player.current_media.album
+            self._attr_media_duration = player.current_media.duration
+            # grab these from the player directly if it supports other sources than MA?
+            self._attr_shuffle = None
+            self._attr_repeat = None
+            if TYPE_CHECKING:
+                assert player.elapsed_time is not None
+            self._attr_media_position = int(player.elapsed_time)
+            self._attr_media_position_updated_at = (
+                from_utc_timestamp(player.elapsed_time_last_updated)
+                if player.elapsed_time_last_updated
+                else None
+            )
+            if TYPE_CHECKING:
+                assert player.elapsed_time is not None
+            self._prev_time = player.elapsed_time
+            return
+
+        if queue is None:
+            # player is completely idle without any source active
+            self._attr_source = player.active_source
+            self._attr_app_id = player.active_source
+            return
+
+        # player has MA as active source (either a group player or the players own queue)
+        self._attr_app_id = DOMAIN
+        self._attr_shuffle = queue.shuffle_enabled
+        self._attr_repeat = queue.repeat_mode.value
+        if not (cur_item := queue.current_item):
+            # queue is empty
+            return
+
+        self._attr_media_content_id = queue.current_item.uri
+        self._attr_media_duration = queue.current_item.duration
+        self._attr_media_position = int(queue.elapsed_time)
+        self._attr_media_position_updated_at = from_utc_timestamp(
+            queue.elapsed_time_last_updated
+        )
+        self._prev_time = queue.elapsed_time
+
+        # handle stream title (radio station icy metadata)
+        if (stream_details := cur_item.streamdetails) and stream_details.stream_title:
+            self._attr_media_album_name = cur_item.name
+            if " - " in stream_details.stream_title:
+                stream_title_parts = stream_details.stream_title.split(" - ", 1)
+                self._attr_media_title = stream_title_parts[1]
+                self._attr_media_artist = stream_title_parts[0]
+            else:
+                self._attr_media_title = stream_details.stream_title
+            return
+
+        if not (media_item := cur_item.media_item):
+            # queue is not playing a regular media item (edge case?!)
+            self._attr_media_title = cur_item.name
+            return
+
+        # queue is playing regular media item
+        self._attr_media_title = media_item.name
+
+        if media_item.media_type == MediaType.TRACK:
+            self._attr_media_artist = getattr(media_item, "artist_str", None)
+            if TYPE_CHECKING:
+                assert isinstance(media_item, Track)
```

Comment:
I am assuming that `Track` is typed and thus if we move this more to the top, we can do `media_item.artist_str` instead of the `getattr`